agent/hcp/telemetry_provider.go

```diff
@@ -69,44 +70,29 @@ func (d *dynamicConfig) equals(newCfg *dynamicConfig) (bool, error) {
 	return currHash == newHash, err
 }
 
-// providerParams is used to initialize a hcpProviderImpl.
-type providerParams struct {
-	metricsConfig   *client.MetricsConfig
-	refreshInterval time.Duration
-	hcpClient       client.Client
-}
-
 // NewHCPProvider initializes and starts a HCP Telemetry provider with provided params.
-func NewHCPProvider(ctx context.Context, params *providerParams) (*hcpProviderImpl, error) {
-	if params.hcpClient == nil {
-		return nil, errors.New("missing HCP client")
-	}
-
-	if params.metricsConfig == nil {
-		return nil, errors.New("missing metrics config")
-	}
-
-	if params.refreshInterval <= 0 {
-		return nil, fmt.Errorf("invalid refresh interval: %d", params.refreshInterval)
-	}
-
-	cfg := &dynamicConfig{
-		Endpoint:        params.metricsConfig.Endpoint,
-		Labels:          params.metricsConfig.Labels,
-		Filters:         params.metricsConfig.Filters,
-		RefreshInterval: params.refreshInterval,
-	}
-
-	ticker := time.NewTicker(params.refreshInterval)
+func NewHCPProvider(ctx context.Context, hcpClient client.Client) *hcpProviderImpl {
+	ticker := time.NewTicker(defaultTelemetryConfigRefreshInterval)
 	t := &hcpProviderImpl{
-		cfg:       cfg,
-		hcpClient: params.hcpClient,
+		cfg: &dynamicConfig{
```

Comment:
Provide sane defaults to avoid nil pointers in the Get* methods - cleaner than nil checks everywhere.